neurom/core/types.py

```diff
@@ -155,107 +156,141 @@ def __ne__(self, other):
+    if len(args) > 1:
+        raise ValueError(f"The {cls.__name__} class constructor accepts only 1 argument.")
 
-    @classmethod
-    def register(cls, name, value):
-        """Register a new value in the Enum class."""
-        new_value = SubtypeCollection(value)
-        new_value_as_tuple = tuple(new_value)
-        err = None
-        if name in cls._member_names_:
-            err = (name, cls._member_map_[name].value)
-        if new_value_as_tuple in cls._value2member_map_:
-            err = (cls._value2member_map_[new_value_as_tuple].name, value)
-        if err is not None:
-            raise ValueError(
-                f"The NeuriteType '{err[0]}' is already registered with the value '{err[1]}'"
-            )
-        obj = super().__new__(cls, new_value)
-        obj._name_ = name
-        obj._value_ = new_value
-        cls._value2member_map_[new_value] = obj
-        cls._member_map_[name] = obj
-        cls._member_names_.append(name)
-        return obj
-
-    @classmethod
-    def unregister(cls, name):
-        """Unregister a value in the Enum class."""
-        if name not in cls._member_names_:
-            raise ValueError(
-                f"The NeuriteType '{name}' is not registered so it can not be unregistered"
-            )
-
-        value = cls._member_map_[name].value
-        del cls._value2member_map_[value]
-        del cls._member_map_[name]
-        cls._member_names_.remove(name)
-
-
-def _enum_accept_undefined(cls, value):
-    # pylint: disable=protected-access
+    value = args[0]
 
     # Use NeuriteType name
-    if isinstance(value, NeuriteType):
+    if isinstance(value, cls):
         value_str = value.name
-        if value_str in cls._member_map_:
-            return cls._member_map_[value_str]
+        if hasattr(cls, value_str):
+            return getattr(cls, value_str)
 
     # Name given as string
     elif isinstance(value, str):
-        if value in cls._member_map_:
-            return cls._member_map_[value]
+        if hasattr(cls, value):
+            return getattr(cls, value)
 
     # Composite type or unhashable type (e.g. list)
     else:
         try:
-            subtype_value = tuple(SubtypeCollection(value))
+            subtype_value = SubtypeCollection(value).subtypes
         except Exception:  # pylint: disable=broad-exception-caught
             pass
         else:
-            if subtype_value in cls._value2member_map_:
-                return cls._value2member_map_[subtype_value]
+            if subtype_value in cls._reverse_mapping:
+                return getattr(cls, cls._reverse_mapping[subtype_value])
 
     # Invalid value
     raise ValueError(f"{value} is not a valid registered NeuriteType")
 
 
-NeuriteType.__new__ = _enum_accept_undefined
+def _empty_init(self, *args, **kwargs):  # pylint: disable=unused-argument
+    """Replace default __init__ to do nothing more than the __new__ method."""
+
+
+class MutableEnumMeta(type):
```

Comment:
Minor: MutableEnumType would be a better choice given that EnumMeta has been renamed to EnumType.
https://github.com/python/cpython/blob/3e97c001711ab68e3d54d65e264cb5c37fefbec0/Lib/enum.py#L1046

Reply:
Good point, I changed